tests/test_generic.py

```diff
@@ -266,11 +249,29 @@ def test_name_object(caplog):
     with pytest.raises(PyPdfError):
         NameObject("/\0").write_to_stream(BytesIO())
 
+    # testing all allowed values
+    NameObject.encoding = "latin1"
```

Comment:
Have you verified that this has no side-effects if the test fails?

Reply:
null char is explicitly forbidden in specs, I can not guarantee that's there is no side effects, but risk seems very low since it's the end of string symbol used in C, so using it could have much more side effects in other implementations. And the risk is only in write_to_stream, not in parsing, furthermore it will create a corrupted file to allow it.

Reply:
This is not about the null character, but IMHO you are changing the class default here without ensuring that a test failure will revert this value correctly without affecting all the other tests.